backend/dataall/modules/dataset_sharing/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -324,29 +330,40 @@ def update_dataset_bucket_key_policy(self):
             'Updating dataset Bucket KMS key policy...'
         )
         key_alias = f"alias/{self.dataset.KmsAlias}"
-        kms_client = KmsClient(account_id=self.source_account_id, region=self.source_environment.region)
+        kms_client = KmsClient(self.source_account_id, self.source_environment.region)
         kms_key_id = kms_client.get_key_id(key_alias)
-        existing_policy = kms_client.get_key_policy(kms_key_id)
-        target_requester_id = SessionHelper.get_role_id(self.target_account_id, self.target_requester_IAMRoleName)
-        if existing_policy and f'{target_requester_id}:*' not in existing_policy:
-            policy = json.loads(existing_policy)
-            policy["Statement"].append(
-                {
-                    "Sid": f"{target_requester_id}",
-                    "Effect": "Allow",
-                    "Principal": {
-                        "AWS": "*"
-                    },
-                    "Action": "kms:Decrypt",
-                    "Resource": "*",
-                    "Condition": {
-                        "StringLike": {
-                            "aws:userId": f"{target_requester_id}:*"
-                        }
-                    }
-                }
-            )
-            kms_client.put_key_policy(kms_key_id, json.dumps(policy))
+        existing_policy = json.loads(kms_client.get_key_policy(kms_key_id))
+        target_requester_arn = self.get_role_arn(self.target_account_id, self.target_requester_IAMRoleName)
```

Comment:
Folder sharing for SSE-S3 encrypted datasets fails because of line 335:
In this case, `existing_policy = kms_client.get_key_policy(kms_key_id)` return `None`. If we apply `json.loads` to this before checking if the policy exists it is going to fail. I think we can move the json.loads inside the if block in line 340 and that should solve the issue

Reply:
In addition, for SSE-S3, existing_policy resolves to None. Which means that it tries to execute the else block in line 356, resulting in failure of folder sharing for SSE-S3. We need to add a check of dataset.imported

Reply:
Let me rephrase it, because I think I was not very clear. We have several conditions to check:
If the dataset is encrypted with SSE-S3 or with KMS.
- If KMS yes, then if the KMS policy exists
     - If yes,....
     - If no, create new policy
- If SSE-S3 - do nothing


Maybe instead of implementing checks inside the function we can modify the call to the function and check in `dataall/modules/dataset_sharing/services/share_processors/s3_access_point_process_share.py:93`

```
if not dataset.imported or dataset.importedKmsKey:
    sharing_folder.update_dataset_bucket_key_policy()
```
And the same applies to `dataall/modules/dataset_sharing/services/share_processors/s3_access_point_process_share.py:212`. It should only be called if KMS is used for encryption.
```
if not dataset.imported or dataset.importedKmsKey:
   clean_up_folder.delete_dataset_bucket_key_policy(
                dataset=dataset
            )
```

Reply:
okay now that makes sense. Addressed in next revision. Please take a look!

Just to confirm, if we create/import a dataset in data all which has SSE-S3 encryption, we wont handle cleaning/updating/removing it via data all correct?